### PR TITLE
Add usage of monitoring with Adam to notebook

### PR DESCRIPTION
**Pull request by sethaxen**
**PR type:** doc improvement

## Summary

**Proposed changes**

* Add example to monitoring notebook describing how to use `monitor` with the `Adam` optimizer

### Release notes

**Fully backwards compatible:** yes

## PR checklist

- [ ] New features: code is well-documented
  - [ ] detailed docstrings (API documentation)
  - [x] notebook examples (usage demonstration)
- [ ] The bug case / new feature is covered by unit tests
- [ ] Code has type annotations
- [ ] Build checks
  - [ ] I ran the black+isort formatter (`make format`)
  - [ ] I locally tested that the tests pass (`make check-all`)
- [x] Release management
  - [x] RELEASE.md updated with entry for this change
  - [x] New contributors: I've added myself to CONTRIBUTORS.md

### CODE REVIEW

File: doc/sphinx/notebooks/getting_started/monitoring.pct.py [hunk 1]
```diff
@@ -246,3 +246,12 @@ def my_image_2(
...
 # Here we needed to redefine `my_scalar` and `my_image` because we had hardcoded the model.
+
+# %% [markdown]
+# `gpflow.optimizers.Scipy.minimize()` runs an optimizer until convergence.
+# Other optimizers, like `tensorflow.keras.optimizers.Adam()` need `.minimize()` to be called in a loop:
+
+opt = tf.keras.optimizers.Adam()
+for i in range(10):
+    opt.minimize(model.training_loss, model.trainable_variables)
+    monitor(i)
```
Comment (st--) [inline doc/sphinx/notebooks/getting_started/monitoring.pct.py:257]:
```suggestion

@tf.function
def optimization_step():
    opt.minimize(model.training_loss, model.trainable_variables)

for i in range(10):
    optimization_step()
    monitor(i)
```

**Comment by st--** [comment]
Thanks! It would be best to include how to use `@tf.function` (as in my code suggestion) so people aren't wondering why their code is so slow ... (it's not consistent everywhere, we should fix this in other notebooks too, but better not to proliferate it:)). Otherwise looks good to me!

**Comment by sethaxen** [comment]
Great! Seems to fail the format-check CI. But when I run `make format` locally it seems to format everything but this file.

**Comment by sethaxen** [comment]
Wait nvm, I had not run the formatter since the PR review.

**Review by st-- — APPROVED** [review]
looks good to me, thanks for contriibuting!

**Pull request merged**